src/inspector_socket.cc

```diff
@@ -550,10 +550,15 @@ static void init_handshake(inspector_socket_t* inspector) {
 int inspector_accept(uv_stream_t* server, inspector_socket_t* inspector,
                      handshake_cb callback) {
   ASSERT_NE(callback, nullptr);
-  CHECK_EQ(inspector->http_parsing_state, nullptr);
+
+  inspector->ws_state = nullptr;
+  inspector->buffer.clear();
+  inspector->ws_mode = false;
+  inspector->shutting_down = false;
+  inspector->connection_eof = false;
```

Comment:
Thank you for the review. I introduced the function... But that pushed the struct to class territory too far for me :)

I changed the struct to class, renamed it appropriately and added the namespace. In fairness, other inspector_* functions need to become member functions as well and the state needs to be made private, etc.

But I am currently working on a complex refactoring that will significantly change this class (I'm hoping to remove libuv interactions) so I don't see much value in doing a comprehensive change right now...

Please take a look at updated CL.